From bc7c0f63b095defe92b7f3f7205e77da63c81168 Mon Sep 17 00:00:00 2001
From: fengmushu <fengmushu@gmail.com>
Date: Mon, 27 Aug 2018 13:00:33 +0800
Subject: [PATCH 1/2] fixup rtc fout clk gate

---
 drivers/rtc/rtc-sunxi.c | 28 +++++++++++++++++++++++++++-
 1 file changed, 27 insertions(+), 1 deletion(-)

diff --git a/drivers/rtc/rtc-sunxi.c b/drivers/rtc/rtc-sunxi.c
index fbefc595e..bbb66760d 100644
--- a/drivers/rtc/rtc-sunxi.c
+++ b/drivers/rtc/rtc-sunxi.c
@@ -38,7 +38,11 @@
 #define SUNXI_LOSC_CTRL				0x0000
 #define SUNXI_LOSC_CTRL_KEY			(0x16aa << 16)
 #define SUNXI_LOSC_CTRL_RTC_HMS_ACC		BIT(8)
-#define SUNXI_LOSC_CTRL_RTC_YMD_ACC		BIT(7)						
+#define SUNXI_LOSC_CTRL_RTC_YMD_ACC		BIT(7)
+#define SUNXI_LOSC_CTRL_AUTO_SWT_EN		BIT(14)
+#define SUNXI_LOSC_CTRL_EXT_SRC	    	BIT(0)
+#define SUNXI_LOSC_CTRL_EXT_GSM 		BIT(4)
+
 
 /* RTC */
 #define SUNXI_RTC_YMD				0x0010
@@ -59,6 +63,11 @@
 #define SUNXI_ALRM1_IRQ_EN			0x0048
 #define SUNXI_ALRM1_IRQ_STA			0x004c
 
+/* Others */
+#define SUNXI_ALARM_CONFIG	    		0x0050
+#define SUNXI_ALRM_WAKEUP_OUTPUT_EN		BIT(0)
+#define SUNXI_LOSC_FOUT_GATE    		0x0060
+
 #define SUNXI_MASK_DH				0x0000001f
 #define SUNXI_MASK_SM				0x0000003f
 #define SUNXI_MASK_M				0x0000000f
@@ -206,6 +215,7 @@ static void sunxi_rtc_setaie(unsigned int to, struct sunxi_rtc_dev *chip)
 
 	writel(alrm_val, chip->base + SUNXI_ALRM_EN);
 	writel(alrm_irq_val, chip->base + SUNXI_ALRM_IRQ_EN);
+	writel(SUNXI_ALRM_WAKEUP_OUTPUT_EN, chip->base + SUNXI_ALARM_CONFIG);
 }
 
 static int sunxi_rtc_getalarm(struct device *dev, struct rtc_wkalrm *wkalrm)
@@ -498,6 +508,22 @@ static int sunxi_rtc_probe(struct platform_device *pdev)
 	writel(SUNXI_ALRM_IRQ_STA_CNT_IRQ_PEND, chip->base +
 			SUNXI_ALRM_IRQ_STA);
 
+	/* init rtc */
+	{
+		unsigned int tmp;
+		tmp = readl(chip->base + SUNXI_LOSC_CTRL);
+		tmp &= (~SUNXI_LOSC_CTRL_AUTO_SWT_EN);
+		tmp |= (SUNXI_LOSC_CTRL_EXT_SRC |
+					SUNXI_LOSC_CTRL_KEY |
+					SUNXI_LOSC_CTRL_EXT_GSM);
+		writel(tmp, chip->base + SUNXI_LOSC_CTRL);
+
+		/* Enable the LOSC fout */
+		dev_info(&pdev->dev, "RTC enable FOUT gate\n");
+		writel(BIT(0), chip->base + SUNXI_LOSC_FOUT_GATE);
+	}
+	device_init_wakeup(&pdev->dev, true);
+
 	chip->rtc = rtc_device_register("rtc-sunxi", &pdev->dev,
 			&sunxi_rtc_ops, THIS_MODULE);
 	if (IS_ERR(chip->rtc)) {
-- 
2.17.1

