diff -urN a/rlt_wifi/os/linux/rt_proc.c b/rlt_wifi/os/linux/rt_proc.c
--- a/rlt_wifi/os/linux/rt_proc.c	2018-08-06 20:58:34.951262596 +0800
+++ b/rlt_wifi/os/linux/rt_proc.c	2018-08-06 20:58:55.639117155 +0800
@@ -36,6 +36,10 @@
 int wl_proc_init(void);
 int wl_proc_exit(void);
 
+#ifdef CONFIG_RALINK_MT7612E
+#define PROCREG_DIR             "mt7612e"
+#endif
+
 #ifdef CONFIG_RALINK_RT2880
 #define PROCREG_DIR             "rt2880"
 #endif /* CONFIG_RALINK_RT2880 */
@@ -61,7 +65,7 @@
 #endif /* PROCREG_DIR */
 
 #ifdef CONFIG_PROC_FS
-extern struct proc_dir_entry *procRegDir;
+struct proc_dir_entry *procRegDir;
 
 #ifdef VIDEO_TURBINE_SUPPORT
 extern BOOLEAN UpdateFromGlobal;
@@ -380,8 +384,9 @@
 	GLOBAL_AP_VIDEO_CONFIG.TxLifeTimeMode = FALSE;
 	GLOBAL_AP_VIDEO_CONFIG.TxLifeTime = 0;
 	GLOBAL_AP_VIDEO_CONFIG.TxRetryLimit = 0;
-
-		proc_ralink_wl = proc_mkdir("wl", procRegDir);
+	if (procRegDir == NULL)
+		procRegDir = proc_mkdir(PROCREG_DIR, NULL);
+	proc_ralink_wl = proc_mkdir("wl", procRegDir);
 
 	if (proc_ralink_wl)
 		proc_ralink_wl_video = proc_mkdir("VideoTurbine", proc_ralink_wl);
