diff -uNr a/rlt_wifi/common/cmm_info.c b/rlt_wifi/common/cmm_info.c
--- a/rlt_wifi/common/cmm_info.c	2018-08-13 20:44:29.379669925 +0800
+++ b/rlt_wifi/common/cmm_info.c	2018-08-13 20:43:51.495396813 +0800
@@ -2147,7 +2147,8 @@
 
 		/*SSID*/
 	NdisZeroMemory(Ssid, (MAX_LEN_OF_SSID +1));
-	if (RTMPCheckStrPrintAble((PCHAR)pBss->Ssid, pBss->SsidLen))
+	//if (RTMPCheckStrPrintAble((PCHAR)pBss->Ssid, pBss->SsidLen))
+	if (1)
 		NdisMoveMemory(Ssid, pBss->Ssid, pBss->SsidLen);
 	else
 	{
