diff -urN a/rlt_wifi/ap/ap_apcli.c b/rlt_wifi/ap/ap_apcli.c
--- a/rlt_wifi/ap/ap_apcli.c	2018-08-21 11:04:21.439682717 +0800
+++ b/rlt_wifi/ap/ap_apcli.c	2018-08-21 11:23:22.983917845 +0800
@@ -3004,6 +3004,8 @@
 	DBGPRINT(RT_DEBUG_TRACE, ("---> ApCliAutoConnectExec()\n"));
 
 	ifIdx = pObj->ioctl_if;
+	if(pAd->ApCfg.ApCliTab[ifIdx].CtrlCurrState == APCLI_CTRL_CONNECTED)
+		return TRUE;
 	CfgSsidLen = pAd->ApCfg.ApCliTab[ifIdx].CfgSsidLen;
 	pCfgSsid = pAd->ApCfg.ApCliTab[ifIdx].CfgSsid;
 	pScanTab = &pAd->ScanTab;
@@ -3087,7 +3089,8 @@
 	if ((pSsidBssTab->BssNr == 0))
 	{
 		DBGPRINT(RT_DEBUG_TRACE, ("No match entry.\n"));
-		pAd->ApCfg.ApCliAutoConnectRunning = FALSE;
+		Set_SiteSurvey_Proc(pAd, pCfgSsid);
+//		pAd->ApCfg.ApCliAutoConnectRunning = FALSE;
 	}
 	else if (pSsidBssTab->BssNr > 0 &&
 			pSsidBssTab->BssNr <=MAX_LEN_OF_BSS_TABLE)
@@ -3189,7 +3192,8 @@
 	else
 	{
 		DBGPRINT(RT_DEBUG_TRACE, ("No candidate AP, the process is about to stop.\n"));
-		pAd->ApCfg.ApCliAutoConnectRunning = FALSE;
+		Set_SiteSurvey_Proc(pAd, pApCliEntry->CfgSsid);
+//		pAd->ApCfg.ApCliAutoConnectRunning = FALSE;
 	}
 	
 	Set_ApCli_Enable_Proc(pAd, "1");
diff -urN a/rlt_wifi/ap/ap_cfg.c b/rlt_wifi/ap/ap_cfg.c
--- a/rlt_wifi/ap/ap_cfg.c	2018-08-21 11:15:09.361647978 +0800
+++ b/rlt_wifi/ap/ap_cfg.c	2018-08-21 11:16:39.033694590 +0800
@@ -11698,7 +11698,7 @@
 		/*      AUTO_SCAN_MODE_FULL
 		use site survey function to trigger auto connecting (when pAd->ApCfg.ApAutoConnectRunning == TRUE)
 		*/
-		Set_SiteSurvey_Proc(pAd, "");//pApCfg->ApCliTab[ifIndex].CfgSsid);
+		Set_SiteSurvey_Proc(pAd, pApCfg->ApCliTab[ifIndex].CfgSsid);//pApCfg->ApCliTab[ifIndex].CfgSsid);
 	}
 
 	return TRUE;
diff -urN a/rlt_wifi/common/cmm_wpa.c b/rlt_wifi/common/cmm_wpa.c
--- a/rlt_wifi/common/cmm_wpa.c	2018-08-21 11:04:21.467686799 +0800
+++ b/rlt_wifi/common/cmm_wpa.c	2018-08-21 11:07:08.479253504 +0800
@@ -1725,7 +1725,7 @@
 				)
 			{
 				DBGPRINT(RT_DEBUG_TRACE, ("Apcli auto connected:PeerPairMsg3Action() \n"));
-				pAd->ApCfg.ApCliAutoConnectRunning = FALSE;
+//				pAd->ApCfg.ApCliAutoConnectRunning = FALSE;
 			}
 #endif /* APCLI_AUTO_CONNECT_SUPPORT */
 #endif /* APLCI_SUPPORT */
diff -urN a/rlt_wifi/common/scan.c b/rlt_wifi/common/scan.c
--- a/rlt_wifi/common/scan.c	2018-08-21 11:17:09.271012722 +0800
+++ b/rlt_wifi/common/scan.c	2018-08-21 11:10:47.459813569 +0800
@@ -482,12 +482,13 @@
 
 		{
 			BOOLEAN bScanPassive = FALSE;
-			if (pAd->MlmeAux.Channel > 14)
+/*			if (pAd->MlmeAux.Channel > 14)
 			{
 				if ((pAd->CommonCfg.bIEEE80211H == 1)
 					&& RadarChannelCheck(pAd, pAd->MlmeAux.Channel))
 					bScanPassive = TRUE;
 			}
+*/
 #ifdef CARRIER_DETECTION_SUPPORT
 			if (pAd->CommonCfg.CarrierDetect.Enable == TRUE)
 				bScanPassive = TRUE;
diff -urN a/rlt_wifi/include/rtmp.h b/rlt_wifi/include/rtmp.h
--- a/rlt_wifi/include/rtmp.h	2018-08-21 11:04:21.463686217 +0800
+++ b/rlt_wifi/include/rtmp.h	2018-08-21 11:06:17.826187490 +0800
@@ -9127,6 +9127,7 @@
 INT Set_ApCli_EncrypType_Proc(RTMP_ADAPTER *pAd, PSTRING arg);
 INT Set_ApCli_Enable_Proc(IN PRTMP_ADAPTER pAd, IN PSTRING arg);
 INT Set_ApCli_Ssid_Proc(IN PRTMP_ADAPTER pAd, IN PSTRING arg);
+INT Set_SiteSurvey_Proc(IN PRTMP_ADAPTER pAd, IN PSTRING arg);
 
 
 #endif /* APCLI_SUPPORT */
